ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java

```diff
@@ -328,9 +338,23 @@ public ObjStoreBrowser() {
      * In other words, this class should be considered as a singleton class. In fact, if two ObjStoreBrowser instances
      * are created, they may interfere with each other.</p>
      */
-   public ObjStoreBrowser(String logDir) {
-        init(logDir);
-        this.objStoreBrowserMBeanName = arjPropertyManager.getObjectStoreEnvironmentBean().getJmxToolingMBeanName();
+   private ObjStoreBrowser(String logDir) {
```

Comment:
We can't just remove the constructor without deprecating it - people could be using this method already in their apps. Is it possible to leave the old way available for apps that want to use that constructor still?

Reply:
My +1 goes to make this (and the other) constructor(s) private. This class should only be used through a static get method

Reply:
My point is removing the constructor for this class will break existing applications. It needs to be planned in to allow time for users to migrate

Reply:
Yes, that is ok. But since it is preferable to use the getInstance instead I would deprecate it and add a comment.

Reply:
Please consider the class is already deprecated and internal. But I am ok to add the deprecation to the method too.

Reply:
I think it's worth separately tracking this deprecation as it's a different use case

Reply:
Sorry @tomjenkinson, I didn't see you were asking to deprecate it first and then remove it later. That works for me as well.

+1 to open a different JBTM (unfortunately)

Reply:
Thanks @tomjenkinson and @jmfinelli . I opened it here: https://issues.redhat.com/browse/JBTM-3974